Core/Core/Planner/CalendarEvent/ViewModel/EditCalendarEventViewModel.swift

```diff
@@ -357,14 +353,77 @@ final class EditCalendarEventViewModel: ObservableObject {
         )
     }
 
-    private func saveAction() -> AnyPublisher<Void, Error>? {
+    private func makeEditConfirmation(event: CalendarEvent) -> ConfirmationViewModel<SeriesModificationType> {
+        guard event.isPartOfSeries else { return .init() }
+
+        return ConfirmationViewModel(
+            title: String(localized: "Confirm Changes", bundle: .core),
+            cancelButtonTitle: String(localized: "Cancel", bundle: .core),
+            confirmButtons: [
+                .init(
+                    title: String(localized: "Change this event", bundle: .core),
+                    option: .one
+                ),
+                .init(
+                    title: String(localized: "Change all events", bundle: .core),
+                    option: .all
+                ),
+                event.isSeriesHead ? nil : .init(
+                    title: String(localized: "Change this and all following events", bundle: .core),
+                    option: .following
+                )
+            ].compactMap { $0 }
+        )
+    }
+
+    private func saveEventAfterConfirmation(
+        on subject: PassthroughSubject<Void, Never>,
+        completion: @escaping (PlannerAssembly.Completion) -> Void
+    ) {
+        subject
+            .map { [weak self] in
+                guard self?.wasEventPartOfSeries == true else { return }
+                self?.shouldShowEditConfirmation = true
+            }
+            .flatMap { [weak self] () -> AnyPublisher<SeriesModificationType?, Never> in
+                guard let self else {
+                    return Empty().eraseToAnyPublisher()
+                }
+                guard wasEventPartOfSeries else {
+                    return Just(nil).eraseToAnyPublisher()
+                }
+                return editConfirmation.userConfirmsOption()
+                    .map { $0 }
+                    .eraseToAnyPublisher()
+            }
+            .map { [weak self] in
+                self?.state = .data(loadingOverlay: true)
+                return $0
+            }
+            .flatMap { [weak self] in
+                (self?.saveAction(seriesModificationType: $0) ?? Empty().eraseToAnyPublisher())
```

Comment:
How about making `saveAction()` to return non-optional publisher, with `Empty()` if model doesn't exists.

```swift
private func saveAction(seriesModificationType: SeriesModificationType?) -> AnyPublisher<Void, Error> {
    guard let model else { return Empty().eraseToAnyPublisher() }
    ...
```

Reply:
Because of the `weak self` we would still need to fallback to `Empty()` here as well. That's why I would keep it simply `nil` in the helper method's `guard`.